lib/si-layer-cache/src/db.rs

```diff
@@ -179,50 +179,81 @@ where
         // Validate configuration
         validate_config(&config)?;
 
-        // Initialize S3 layers if mode requires it
-        let s3_layers = if config.persister_mode != PersisterMode::PostgresOnly {
-            use crate::s3::KeyTransformStrategy;
-
-            let mut layers = HashMap::new();
-
-            // Define cache configurations with their strategies
-            let cache_configs = [
-                (cas::CACHE_NAME, KeyTransformStrategy::Passthrough),
-                (change_batch::CACHE_NAME, KeyTransformStrategy::Passthrough),
-                (
-                    workspace_snapshot::CACHE_NAME,
-                    KeyTransformStrategy::Passthrough,
-                ),
-                (rebase_batch::CACHE_NAME, KeyTransformStrategy::Passthrough),
-                (
-                    encrypted_secret::CACHE_NAME,
-                    KeyTransformStrategy::Passthrough,
-                ),
-                (func_run::CACHE_NAME, KeyTransformStrategy::ReverseKey), // ULID-based
-                (func_run_log::CACHE_NAME, KeyTransformStrategy::ReverseKey), // ULID-based
-                (
-                    split_snapshot_subgraph::CACHE_NAME,
-                    KeyTransformStrategy::Passthrough,
-                ),
-                (
-                    split_snapshot_supergraph::CACHE_NAME,
-                    KeyTransformStrategy::Passthrough,
-                ),
-                (
-                    split_snapshot_rebase_batch::CACHE_NAME,
-                    KeyTransformStrategy::Passthrough,
-                ),
-            ];
-
-            for (cache_name, strategy) in cache_configs {
-                let cache_config = config.object_storage_config.for_cache(cache_name);
-                let s3_layer = S3Layer::new(cache_config, strategy)?;
-                layers.insert(cache_name, s3_layer);
-            }
+        // Always create S3 layers for health check validation
+        use crate::s3::KeyTransformStrategy;
+        use telemetry::prelude::*;
```

Comment:
nit: since this is no longer in the if, should we just move these imports to the top?